### PR TITLE
Update for Orleans 1.5.0

### PR DESCRIPTION
**Pull request by pditterline**
Upgraded packages to Orleans 1.5.0, StackExchange.Redis 1.2.4
Changed to non-static serialization manager
Added explicit client initialization to unit tests

### CODE REVIEW

**Comment by pditterline** [comment]
This is a quick update to get the package working with the breaking changes to SerializationManager in Orleans 1.5.0. I did not update the unit tests to use the new TestCluster, so that still needs to be done, but the tests function with the old (soon-to-be desprecated) test classes and the provider itself is working with 1.5.0.

... it looks like this failed because it couldn't push the nuget package somewhere? Not sure what the CI build is doing.

File: Orleans.StorageProviders.RedisStorage.Tests/DevTestServerConfigurationBinaryFormat.xml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 <OrleansConfiguration xmlns="urn:orleans">
   <Globals>
     <StorageProviders>
-      <Provider Type="Orleans.StorageProviders.RedisStorage" Name="REDIS" RedisConnectionString="localhost" UseJsonFromat="false" />
-      <Provider Type="Orleans.StorageProviders.RedisStorage" Name="PubSubStore" RedisConnectionString="localhost" UseJsonFromat="false" />
+      <Provider Type="Orleans.StorageProviders.RedisStorage" Name="REDIS" RedisConnectionString="auspatlinux2" UseJsonFromat="false" />
```
Comment (pditterline) [inline Orleans.StorageProviders.RedisStorage.Tests/DevTestServerConfigurationBinaryFormat.xml]: bah... this was not supposed to be submitted. I'll take the changes out

**Comment by richorama** [comment]
I think nuget was having an issue...

**Comment by pditterline** [comment]
Anything I/we need to do here to get the nugets to work?

**Comment by richorama** [comment]
I'll check the app veyor configuration. It might not be set up...

**Comment by richorama** [comment]
It looks like it's been published to nuget by someone else: https://www.nuget.org/packages/Orleans.StorageProvider.Redis

I have contacted them.

**Pull request merged**